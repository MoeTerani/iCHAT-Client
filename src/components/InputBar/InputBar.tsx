import React from 'react';
import './InputBar.css';
import { MdSend } from 'react-icons/md';

interface Props {
  room: string;
  message: string;
  setMessage: (message: string) => void;
  sendMessage: (e: any) => void;
}
const InputBar = ({ room, message, setMessage, sendMessage }: Props) => {
  return (
    <form className='form--bar'>
      <input
        className='input--bar'
        type='text'
        placeholder={`Message ${room}...`}
        value={message}
        onChange={({ target: { value } }) => setMessage(value)}
        onKeyPress={(event) =>
          event.key === 'Enter' ? sendMessage(event) : null
        }
      />
      <button className='sendButton' onClick={(e) => sendMessage(e)}>
        <MdSend className='sendIcon' />
      </button>
    </form>
  );
};

export default InputBar;
