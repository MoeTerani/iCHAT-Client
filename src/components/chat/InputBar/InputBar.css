.form--bar {
  display: flex;
  /* border: 1px solid #848688; */
}

.input--bar {
  border: none;
  border-radius: 0;
  padding: 5%;
  width: 85%;
  font-size: 1.2em;
  background-color: #222529;
  color: white;
}

input--bar:focus,
textarea:focus,
select:focus {
  outline: none;
}
textarea:focus,
input:focus,
button:focus {
  outline: none;
}

.sendButton {
  color: #fff !important;
  background: #222529;
  padding: 1px;
  display: inline-block;
  border: none;
  width: 15%;
  cursor: pointer;
}
.sendButton:hover {
  color: #248a52 !important;
  background: #248a52;
}
.sendIcon {
  width: 25px;
  height: 25px;
}
