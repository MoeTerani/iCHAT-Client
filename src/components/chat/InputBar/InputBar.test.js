import { shallow } from 'enzyme';
import React from 'react';
import InputBar from './InputBar';

describe('InputBar components', () => {
  test('it renders without errors', () => {
    expect(shallow(<InputBar />)).toMatchSnapshot();
  });
});
