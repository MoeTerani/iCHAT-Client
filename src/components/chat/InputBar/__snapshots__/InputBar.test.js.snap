// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBar components it renders without errors 1`] = `ShallowWrapper {}`;
